helm/resource_release.go

```diff
@@ -507,6 +512,76 @@ func resourceReleaseCreate(d *schema.ResourceData, meta interface{}) error {
 	return setIDAndMetadataFromRelease(d, rel)
 }
 
+func getHash(v interface{}) (uint64, error) {
+	hash, err := hashstructure.Hash(v, nil)
+	if err != nil {
+		return 0, err
+	}
+	return hash, nil
+}
+
+func getDeepHashChart(charts []*chart.Chart) (uint64, error) {
+	var checkSum uint64
+
+	for _, c := range charts {
+		res, err := getHashChart(c)
+		if err != nil {
+			return checkSum, err
+		}
+
+		checkSum += res
+	}
+
+	return checkSum, nil
+}
+
+func getHashChart(c *chart.Chart) (uint64, error) {
+	var checkSum uint64
+	var res uint64
+	var err error
+	itemToHash := []interface{}{c.Values, c.Templates, c.Metadata, c.Files}
+
+	if len(c.Dependencies()) > 0 {
+		res, err = getDeepHashChart(c.Dependencies())
+		if err != nil {
+			return 0, err
+		}
+		checkSum += res
+	}
+
+	for _, i := range itemToHash {
+		res, err = getHash(i)
+		if err != nil {
+			return 0, err
+		}
+
+		checkSum += res
+	}
+
+	return checkSum, nil
+}
+
+func getLocalHashChart(chartParam interface{}) (string, error) {
```

Comment:
can you move this function and `getHashChart` to the end of the document where are being used?